apps/src/code-studio/components/ExportDialog/Dialog.jsx

```diff
@@ -290,13 +290,19 @@ class ExportDialog extends React.Component {
       });
       return exportResult;
     } catch (e) {
-      recordExport('publishToExpoError');
+      const hasDataAPIsError = e.message.includes('hasDataAPIs');
+      const exportError = hasDataAPIsError
+        ? 'This project uses data APIs. Exporting this type of app is not supported during the Beta period.'
+        : 'Failed to create app. Please try again later.';
+      recordExport(
+        hasDataAPIsError ? 'publishBlockedDueToDataAPIs' : 'publishToExpoError'
+      );
```

Comment:
We will now distinguish this case so we can use the firehose data to determine how often project exports are being blocked for this reason.